app/javascript/components/challenges/Cards.tsx

```diff
@@ -37,28 +66,27 @@ export default function Cards() {
     <div style={{ textAlign: 'center' }}>
       <h1>🥇 Challenge 3</h1>
       <a href="/thanks">Click here when you're finished</a>
-      {/* NASA API docs here: https://api.nasa.gov/ */}
       <h3>Slider</h3>
       <h3>1. Refactor this code to remove duplication and make it more 'Reacty'.</h3>
       <h3>2. Convert the images into a slider using the pagination buttons.</h3>
+
       <div className="cards">
-        <div className="card" style={{ backgroundImage: `url(${image1Url})` }} />
-        <div className="card" style={{ backgroundImage: `url(${image2Url})` }} />
-        <div className="card" style={{ backgroundImage: `url(${image3Url})` }} />
-        <div className="card" style={{ backgroundImage: `url(${image4Url})` }} />
-      </div>
-      <div style={{ display: 'flex', justifyContent: 'center' }}>
-        <button style={buttonStyles}>Previous</button>
-        <button style={buttonStyles}>Next</button>
+        <div className="wrapper">
+          <Carousel className="carousel" initialStep={1} images={images} />
+        </div>
       </div>
+
       <h3>Randomised Image</h3>
       <h3>1. Randomise the image when you click the button.</h3>
+//frustrating part here is I've generated the random aspect, just can't work out
+// how to deal with the onClick
+
       <div className="cards">
-        <div className="card" style={{ backgroundImage: `url(${image1Url})` }} />
-      </div>
-      <div style={{ display: 'flex', justifyContent: 'center' }}>
-        <button style={buttonStyles}>Randomise</button>
+          <div className="card" style={{ backgroundImage: `${baseUri}?api_key=${nasaApiKey}&date=${myDate}` }} /></div>
+            <div style={{ display: 'flex', justifyContent: 'center' }}>
+            <button style={buttonStyles} >Randomise</button>
```

Comment:
So for the onClick add an onclick handler to the button;

`<button style={buttonStyles} onClick={handleRandomise}>Randomise</button>`

Then we also need to do something with the onClick so:

```
function handleRandomise() {
   const randomDate = "2019-0"+Math.floor(Math.random()*12).toString() +"-" +  
   Math.floor(Math.random()*29).toString();
     getImage(randomDate).then(response => setRandomImage(response))
   }
```

Then we need some new state `const [randomImage, setRandomImage] = React.useState('')`

And finally change the background image for the card:

```
<div className="card" style={{ backgroundImage: `url(${randomImage})` }} /></div>
  <div style={{ display: 'flex', justifyContent: 'center' }}>
    <button style={buttonStyles} onClick={handleRandomise}>Randomise</button>
  </div>
```